cli/yamls.py

```diff
@@ -330,29 +57,126 @@ def to_yaml(cls, dumper, node):
         raise exceptions.IRPlaceholderException(message)
 
 
-def load(file_path, update_placeholders=True):
+def load(settings_file, update_placeholders=True):
     """
-    Loads the yaml file and return it dict representation.
+    Loads and returns the content of a given YAML file as a dictionary
+
+    :param settings_file: String represents a path to a YAML file
+    :param update_placeholders: Whether to update Placeholder list or not in
+    case of Placeholder tag is in the given YAML file
+    :return: Dictionary containing the content the given YAML file
     """
-    LOG.debug("Loading setting file: %s" % file_path)
-    if not os.path.exists(file_path):
-        raise exceptions.IRFileNotFoundException(file_path)
+    LOG.debug("Loading setting file: %s" % settings_file)
+    if not os.path.exists(settings_file):
+        raise exceptions.IRFileNotFoundException(settings_file)
 
     try:
-        res = dict(configure.Configuration.from_file(file_path).configure())
+        with open(settings_file) as f_obj:
+            loaded_yml = yaml.load(f_obj)
+
+        # Handling case of empty file
+        if loaded_yml is None:
+            raise exceptions.IREmptySettingsFile(settings_file)
+
         if update_placeholders:
-            placeholders_list = Placeholder.placeholders_list
-            for placeholder in placeholders_list[::-1]:
+            for placeholder in Placeholder.placeholders_list[::-1]:
                 if placeholder.file_path is None:
-                    placeholder.file_path = file_path
+                    placeholder.file_path = settings_file
                 else:
                     break
-        return res
+
+        return loaded_yml
+
     except yaml.constructor.ConstructorError as e:
-        raise exceptions.IRYAMLConstructorError(e, file_path)
-    except configure.ConfigurationError as confError:
-        # fix for the empty yml files
-        if confError.message == 'unconfigured':
-            return {}
-        else:
-            raise
+        raise exceptions.IRYAMLConstructorError(e, settings_file)
+
+
+def _get_most_nested_lookups(string_value):
+    """
+    Returns list of most nested lookups patterns from a given string (in
+    case of lookup in lookup)
+
+    :param string_value: String to search lookup patterns in it
+    :return: List containing lookup patterns
+    """
+    parser = re.compile('\{\{\s*\!lookup\s*[\w.]*\s*\}\}')
+    return parser.findall(string_value)
+
+
+def replace_lookup(lookups_dict):
+    """
+    Replaces all lookup pattern in a given dictionary (lookups_dict) with
+    the corresponding values
+
+    :param lookups_dict: Dictionary with data to replace
+    """
+    def yaml_walk(yaml_content, key_path=None):
```

Comment:
Please remove this inner method as there is no extra logic in having it as an inned method